monai/data/wsi_reader.py

```diff
@@ -418,3 +420,131 @@ def get_patch(
             patch = patch[:3]
 
         return patch
+
+
+@require_pkg(pkg_name="openslide")
+class OpenSlideWSIReader(BaseWSIReader):
+    """
+    Read whole slide images and extract patches using OpenSlide library.
+
+    Args:
+        level: the whole slide image level at which the image is extracted. (default=0)
+            This is overridden if the level argument is provided in `get_data`.
+        kwargs: additional args for `openslide.OpenSlide` module.
+
+    """
+
+    supported_suffixes = ["tif", "tiff", "svs"]
+
+    def __init__(self, level: int = 0, **kwargs):
+        super().__init__(level, **kwargs)
+
+    @staticmethod
+    def get_level_count(wsi) -> int:
```

Comment:
I don't think that we can add type hint for wsi since its types are different depending on using openslide or cucim, and during type checking OpenSlide or CuImge might not be available and will lead to failure of github checks.